CONTRIBUTING.md

```diff
@@ -1,49 +1,122 @@
-| `release/vX.Y.Z` | NO         | `dev`              | A temporary release branch that follows the [semver](http://semver.org/) versioning. This is what is sent to UAT.<br/>A pull request is required to merge code into any `release/vX.Y.Z` branch.   |
-| `hotfix/*`       | NO         | `main`             | These are bug fixes against production.<br/>This is used because develop might have moved on from the last published state.<br/>Remember to merge this back into develop and any release branches. |
+### 1. Creating Feature #1 (Developer #1)
 
+#### Step 1: Clone the Repository
+Clone the repository to your local machine.
+```bash
+git clone git@github.com:aws-solutions-library-samples/osml-imagery-toolkit.git
+```
+
+#### Step 2: Create a New Branch
+Create a new branch off `main`, naming it related to the work being done.
+```bash
+git checkout -b feature/<feature_id>
+```
+
+#### Step 3: Develop and Commit Changes
+On the new branch, code and add commits as necessary.
+
+#### Step 4: Push the New Branch
+Push the new branch to trigger unit tests, static code analysis, Sonar cube checks, security checks, etc.,
+before merging to `main`.
+```bash
+git push origin feature/<feature_id>
+```
+
+#### Step 5: Create a Pull Request (PR)
+Open a PR against `main` to kick off a discussion.
+Open a PR as soon as possible, even if it's not ready (mark it as WIP).
+This ensures ample time for review and discussion, improving code quality.
+
+#### Step 6: PR Review and Merge
+After your PR is reviewed, make any necessary changes by repeating steps 2 and 3.
+A team member will merge your PR after approval,
+ensuring it's reviewed by someone who has not contributed to the branch.
+
+### 2. Creating Feature #2 (Developer #2) and Rebasing
+
+#### Step 1: Update Local Main Branch
+Pull and rebase the remote `main` onto your local `main` branch to keep it up to date.
+```bash
+git pull --rebase origin main
+```
+
+#### Step 2: Develop New Functionality
+Add new functionality commits to your branch.
+
+#### Step 3: Push Changes
+Push your code up to GitHub.
+
+#### Step 4: Perform Tests
+Ensure to perform any necessary tests to validate the new functionality.
+
+#### Step 5: Open a PR
+Open a PR for review. Note that PR may show merge conflicts with `main` due to concurrent feature merges.
+
+#### Step 6: Rebasing
+First, update your local `main` branch.
+```bash
+git checkout main
+git pull --rebase origin main
+```
+Then, rebase your feature branch with `main` to integrate the latest changes.
+```bash
+git checkout <feature-2-branch>
+git rebase main
+```
+Resolve any conflicts during the rebase.
+After rebasing, re-test your branch, make necessary changes,
+and push to the remote with the `-f` flag to overwrite history.
+```bash
+git push origin <feature-2-branch> -f
+```
+
+### Creating a Release
```

Comment:
I think we should delete this from here and move it into our own internal documentation. Contributors shouldn't be making releases.

Reply:
Done